components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/cache/PolicyCache.java

```diff
@@ -138,30 +138,34 @@ public void invalidateCache() {
      *
      * @return
      */
-    public boolean isInvalidate(){
+    public boolean isInvalidate() {
 
         int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
-        int state = 0 ;
+        int state = 0;
 
         synchronized (cacheInvalidationState) {
-            if(cacheInvalidationState.get(tenantId)!=null) {
+            if (cacheInvalidationState.get(tenantId) != null) {
                 state = cacheInvalidationState.get(tenantId);
+            } else {
+                // we ignore the case where the cache invalidation state is not present.This means the cache is valid.
             }
-
-            cacheInvalidationState.put(tenantId,0);
-        }
-        if(log.isDebugEnabled()){
-            log.debug("Check the invalidation state of all cache : " + state);
         }
 
-        if(state==1){
-            return true ;
-        }else{
-            return false ;
+        boolean isInvalid = state == 1;
```

Comment:
For better readability please change, 
boolean isInvalid = state == 1;
-- > boolean isInvalid = (state == INVALID_STATE);
and have a constant for INVALID_STATE